eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.naming.AccessModifier;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ *  No external dependency exposed in the public API
+ */
+public class ExternalDependencyExposedCheck extends AbstractCheck {
+    private static final String JAVA = "java";
+    private static final String COM_AZURE = "com.azure";
+    private static final String REACTOR = "reactor";
+    private static final String DOT = ".";
+
+    private static final String EXTERNAL_DEPENDENCY_ERROR =
+        "Class ''%s'', is a class from external dependency. You should not use it as a return or parameter argument type";
+
+    private static final Set<String> VALID_DEPENDENCY_PACKAGE_NAMES = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+        JAVA, COM_AZURE, REACTOR
+    )));
+
+    private Map<String, String> classPathMap = new HashMap<>();
```

Comment:
private final

Reply:
These aren't actually paths. It's a mapping of the simple class name to its qualified class name. The naming of this variable is deceiving. (ie. simpleClassNameToQualifiedNameMap)

The name threw me off because I was expecting an actual path like C:\users\com\azure\EventHubProducer.java.